### PR TITLE
feat: add requirement_set api

### PR DESCRIPTION
**Pull request by DudeUnleashed**
# Description

Created entities RequirementSet. This will allow access to various information relating to requirement sets for the Courseflow component of OnTrack. Various GET, POST, PUT and DELETE endpoints created and implemented, with appropriate tests.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] With the API running, cd into doubtfire-api
- [x] Run the following command: bundle exec rails test test/api/courseflow/requirement_set_api_test.rb 
- [x] Can manually test endpoints using postman.

image

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation if appropriate
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have created or extended unit tests to address my new additions
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -10,7 +10,11 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
+<<<<<<< HEAD
```
Comment (satikaj) [inline db/schema.rb]: These are git diff comments and must be removed

Comment (DudeUnleashed) [reply db/schema.rb]: Thanks, i forgot to remove these for them all after rebasing to 9.x, i went through and fixed them

**Comment by sammyrolfe** [comment]
All tests run successfully and appropriate responses appear from the endpoints when testing in postman. Nice work!

**Comment by sammyrolfe** [comment]
All endpoints requiring authentication.

**Comment by aNebula** [comment]
@DudeUnleashed Is this still relevant this semester and needed after the current changes in direction for courseflow?
If so, please open a PR against https://github.com/doubtfire-lms/doubtfire-api/tree/new/course-flow

**Pull request closed without merging**